src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -198,6 +197,8 @@ object TaskBuilder {
     app.args.foreach { argv =>
       builder.setShell(false)
       builder.addAllArguments(argv.asJava)
+      //mesos containerizer expects the cmd and arguments
+      app.container.getOrElse(builder.setValue(argv.head))
```

Comment:
I don't like that you use `.getOrElse` for a side effect. I would like `if(app.container.isEmpty)` better. 

It is also hard for me to tell what the special case is. Is docker the special case or the command executor? What would you pass to a custom executor? Or another container executor?

But maybe that's to theoretical and we should focus on fixing this ;)